### ISSUE TITLE
Don't pretend `Int` doesn't wrap when incrementing it at `maxBound`.

### ISSUE THREAD

**Issue by chowells79** [ucsd-progsys/liquidhaskell#377]
The following bit of code passes liquid's check:

    {-@ ax1 :: Int -> TRUE @-}
    ax1 :: Int -> Bool
    ax1 x = x < x + 1

But when tested in ghci:

    *Main> ax1 maxBound
    False

There are a lot of cases where the wraparound behavior actually matters, either causing bugs or being necessary for correct behavior. In both cases, the model being checked needs to include that behavior.

**Comment by ranjitjhala** [comment]
Yes this is a tricky one; perhaps we should have a "--bounded-int"
or strict mode for this. It's tricky because it will massively clutter type signatures for users so not sure what the usability trade off is but certainly a "strict" mode would be a first step. Thanks!

**Comment by ranjitjhala** [comment]
Also would be extremely grateful for links to libraries/apps where wraparound matters as they'd give us useful starting points for thinking about how to address this issue in a tractable manner.

Thanks!
